### ISSUE TITLE
Is there a plan to release a dual speaker fine-tuning script?

### ISSUE THREAD

**Issue by RonanKMcGovern** [kyutai-labs/moshi-finetune#6]
### Due diligence

- [x] I have done my due diligence in trying to find the answer myself.

### Topic

The PyTorch implementation

### Question

I just need to adjust the annotation script to correctly timestamp, correct?

Also, what are the default labels for the speakers in the moshiko and moshika models? Thanks

**Comment by LaurentMazare** [comment]
Not sure what you mean exactly by "dual speaker fine-tuning script". The current code assumes that you have separate audio for each speaker, the first channel being the assistant and the second the user. If you only have a single audio channel with both speakers, you may want to use a diarization model, e.g. pyannote to annotate the audio with the speaker identities, and cut the audio accordingly to create two channels. We only finetune for the assistant generation at the moment - I don't think we have open-source weights at the moment that could generate both the assistant and the user if that's what you meant.

As for the default label, you can see in the `annotate.py` script that we use `SPEAKER_MAIN` in the transcript, and that's also what the training code would need (see [finetune.py](https://github.com/kyutai-labs/moshi-finetune/blob/c151df7eedc7e16cc273a7420d7bfd7aecf84554/finetune/data/interleaver.py#L84)).

**Comment by RonanKMcGovern** [comment]
Many thanks, that clarifies things for me. My question didn't make sense.

I was confused as I only saw the SPEAKER_MAIN within the sample dataset [here](https://huggingface.co/datasets/kyutai/DailyTalkContiguous/blob/main/data_stereo/0.json).

But now I think I understand that is because only main speaker (moshi) has got text tokens in the Inner Monologue.
